presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPlugin.java

```diff
@@ -13,31 +13,15 @@
  */
 package io.prestosql.plugin.jdbc;
 
-import com.google.common.collect.ImmutableList;
 import com.google.inject.Module;
-import io.prestosql.spi.Plugin;
-import io.prestosql.spi.connector.ConnectorFactory;
-
-import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Strings.isNullOrEmpty;
-import static java.util.Objects.requireNonNull;
+import com.google.inject.util.Modules;
+import io.prestosql.plugin.jdbc.credential.CredentialProviderModule;
 
 public class JdbcPlugin
-        implements Plugin
+        extends BaseJdbcPlugin
 {
-    private final String name;
-    private final Module module;
-
     public JdbcPlugin(String name, Module module)
     {
-        checkArgument(!isNullOrEmpty(name), "name is null or empty");
-        this.name = name;
-        this.module = requireNonNull(module, "module is null");
-    }
-
-    @Override
-    public Iterable<ConnectorFactory> getConnectorFactories()
-    {
-        return ImmutableList.of(new JdbcConnectorFactory(name, module));
+        super(name, Modules.combine(new CredentialProviderModule(), module));
```

Comment:
Another though. Maybe instead of putting this module here and having `BaseJdbcPlugin` we could add a factory method to `JdbcPlugin`, like
```
JdbcPlugin.createWithCredentials(name, additionalModue);
```
?

What do you think? I have a feeling that using inheritance is too heavy for this.

Reply:
Other plugins are subclasses of JdbcPlugin, how would we initialize them? If a factory method was responsible for adding `CredentialProviderModule` then we wouldn't be able to use super().